From 77009988e4b04cafb082d4e7e84505a6c329ad5b Mon Sep 17 00:00:00 2001
From: LeoLiu-oc <LeoLiu-oc@zhaoxin.com>
Date: Tue, 7 Jun 2022 21:58:56 +0800
Subject: [PATCH] xhci: Fix a logic issue when display Zhaoxin XHCI root hub
 speed
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 77009988e4b04cafb082d4e7e84505a6c329ad5b
Modified-by-SEL: No


zhaoxin inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I40QDN
CVE: NA

----------------------------------------------------------------

Fix a logic issue when display Zhaoxin XHCI root hub speed.

Signed-off-by: LeoLiu-oc <LeoLiu-oc@zhaoxin.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Reviewed-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/usb/host/xhci.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/usb/host/xhci.c b/drivers/usb/host/xhci.c
index ba2b2b5afdd1..0984bd31e0e9 100644
--- a/drivers/usb/host/xhci.c
+++ b/drivers/usb/host/xhci.c
@@ -5271,10 +5271,10 @@ int xhci_gen_setup(struct usb_hcd *hcd, xhci_get_quirks_t get_quirks)
 					if (XHCI_EXT_PORT_PSIV(xhci->port_caps[j].psi[i]) >= 5)
 						minor_rev = 1;
 				}
-				if (minor_rev != 1) {
-					hcd->speed = HCD_USB3;
-					hcd->self.root_hub->speed = USB_SPEED_SUPER;
-				}
+			}
+			if (minor_rev != 1) {
+				hcd->speed = HCD_USB3;
+				hcd->self.root_hub->speed = USB_SPEED_SUPER;
 			}
 		}
 
-- 
2.34.1

